src/EFCore.PG/Infrastructure/Internal/NpgsqlOptionsExtension.cs

```diff
@@ -329,6 +350,28 @@ public class UserRangeDefinition
             SubtypeName = subtypeName;
         }
 
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hashCode = RangeName.GetHashCode();
+                hashCode = (hashCode * 397) ^ (SchemaName != null ? SchemaName.GetHashCode() : 0);
+                hashCode = (hashCode * 397) ^ SubtypeClrType.GetHashCode();
+                hashCode = (hashCode * 397) ^ (SubtypeName != null ? SubtypeName.GetHashCode() : 0);
+                return hashCode;
+            }
+        }
+
+        public override bool Equals(object obj) => obj is UserRangeDefinition urd && Equals(urd);
+
+        public bool Equals(UserRangeDefinition other)
+            => ReferenceEquals(this, other) ||
+               !(other is null) &&
+               string.Equals(RangeName, other.RangeName) &&
+               string.Equals(SchemaName, other.SchemaName) &&
+               SubtypeClrType == other.SubtypeClrType &&
+               string.Equals(SubtypeName, other.SubtypeName);
```

Comment:
All string comparisons could be simplified using the equality operator.

Reply:
You're right, just had these autogenerated and didn't look to closely.